src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTests.cs

```diff
@@ -96,6 +96,12 @@ private static void ValidateAndRoundtrip(object obj, TypeSerializableValue[] blo
                     BinaryFormatterHelpers.ToBase64String(obj, FormatterAssemblyStyle.Full));
             }
 
+            // Check if the passed in value in a serialization entry is assignable by the passed in type.
+            if (!PlatformDetection.IsFullFramework && obj is ISerializable customSerializableObj)
```

Comment:
Rather than disabling it on netfx, should we just disable it for the (one?) cases that won't work? e.g.
```C#
if (obj is ISerializable customSerializableObj && HasObjectTypeIntegrity(customSerializableObj) { ... }
....
private static bool HasObjectTypeIntegrity(ISerializable serializable) =>
    !PlatformDetection.IsFullFramework ||
    !NotFiniteNumberException;
```
or something like that?